syntax = "proto3";

import "google/protobuf/empty.proto";

package configuration;

service DicaService {
    //Recebe o nome dos jogadores
    rpc CriarJogador(NomeJogador) returns (NomeJogadorResp) {};

    //Define a palavra que será adivinhada na partida
    rpc EscolherPalavra(Palavra) returns (PalavraResp){};

    //O jogador que fornece a dica deve receber a palavra
    rpc VerPalavra(google.protobuf.Empty) returns (PalavraResp) {};

    // envia a dica do jogador 
    rpc VerDica(google.protobuf.Empty) returns (Dica) {};

    // o jogador define a dica e retorna o nome do jogador que realizará o palpite
    rpc DarDica(Dica) returns (NomeJogadorResp) {};

    // recebe o palpite do jogador
    rpc DarPalpite(Palpite) returns (google.protobuf.Empty) {};

    // mostra o palpite do jogador
    rpc VerPalpite(google.protobuf.Empty) returns (PalpiteResposta) {};

    // loop da partida
    rpc PartidaStream(Palpite) returns (NomeJogadorResp) {};

    // loop da rodada
    rpc RodadaStream(google.protobuf.Empty) returns(NomeJogadorResp) {};
}

message NomeJogador {
    string nome = 1;
}

message NomeJogadorResp {
    string nome = 1;
    bool recebida = 2;
}

message Palavra {
    string palavra = 1;
}

message PalavraResp {
    string palavra = 1;
    bool recebida = 2;
}

message Dica {
    string dica = 1;
}

message Palpite {
    string palpite = 1;
}

message RodadaResposta {
    bool inicia = 1;
}


message PalpiteResposta {
    string palpite = 1;
    bool acertou = 2;
    bool recebeu = 3;
}